### PR TITLE
=doc add release notes and migration guide for 10.1.0

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+### Depend on akka-stream explicitly
+
+Starting from Akka Http 10.1.0, an explicit dependency to the akka-stream module is necessary. See
+@ref[Compatibility with Akka](../compatibility-guidelines.md#compatibility-with-akka) for more information. The minimum
+Akka version currently required is Akka 2.5.11.
+
+### Removal of deprecated methods
+
+When still compiling with 10.0.11 make sure not to refer to deprecated methods. The deprecation notices usually give hints
+about alternatives.
```
Comment (jlprat) [inline docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md]: Maybe we can add a note that removed methods are the ones prior to 10.0.11.

File: docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## General Notes
+
+See the general @ref[compatibility guidelines](../compatibility-guidelines.md).
+
+## Akka HTTP 10.0.11 - > 10.1.0
+
+### Depend on akka-stream explicitly
+
+Starting from Akka Http 10.1.0, an explicit dependency to the akka-stream module is necessary. See
```
Comment (jlprat) [inline docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md]: `akka-streams` between backticks maybe?

File: docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md (hunk 1 above)
Comment (jrudolph) [reply docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md]: Yep, good point.

File: docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md (hunk 2 above)
Comment (jrudolph) [reply docs/src/main/paradox/migration-guide/migration-guide-10.1.x.md]: :+1:

**Pull request merged**